content/docs/guides/scim/azuread.md

```diff
@@ -0,0 +1,116 @@
+menu:
+    userguides:
+        parent: scim
+        weight: 1
+---
+
+This document outlines the steps required to configure automatic provisioning/deprovisioning of your users in Pulumi using SCIM 2.0.
+
+Please note that some advanced SCIM features aren't supported yet. For more information, see [Known Limitations]({{< relref "#known-limitations" >}}).
+
+## Prerequisites
+
+* Your organization must already be configured to use [SAML SSO]({{< relref "/docs/guides/saml/aad" >}}) with Pulumi.
+* You must be an admin of your Pulumi organization.
+* (Optional, but highly recommended) You should have more than one admin for your Pulumi organization.
+
+## Enabling Automatic Provisioning
+
+1. Navigate to the Azure Active Directory where you have configured Single Sign On using SAML with Pulumi.
+2. Click **Enterprise Applications** and select the app in which you configured Single Sign On with Pulumi earlier.
+3. Click on the **Provisioning** feature, and change the value of **Provisioning Mode** to **Automatic**.
+
+### Admin Credentials
+
+Under the **Admin Credentials** section of the **Provisioning** feature, fill out the form as follows:
+
+* **Tenant URL**: `https://api.pulumi.com/scim/v2/{orgName}`, where `{orgName}` must be replaced with your organization’s login name (not display name). If you do not know this, navigate to your SAML settings and look at the SSO URL. It will have your organization’s login name in the URL.
+* **Secret Token**: You will use a token from the [Pulumi Console](https://app.pulumi.com) as the authorization bearer token. To generate a token, navigate to your org in the Pulumi Console, click on the **Settings** tab, and then click **SAML SSO**. Scroll down to the **SCIM** section and generate a new token if you have never generated one for your org, or regenerate it if you have already done so in the past.
+
+    {{% notes "info" %}}
+Once you generate the token, please save it securely. Neither the Pulumi Console nor Pulumi support can retrieve a token once it's been initially generated. If you lose the SCIM token and need it again, you'll have to generate a new token, which invalidates any previous tokens for your Pulumi organization.
+    {{% /notes %}}
+
+Click **Test Connection**. You should get a success notification once the connection is successful.
+
+### Mappings
+
+Make sure the **Provision Azure Active Directory _Users_** mapping is enabled.
+
+    {{% notes "info" %}}
+If you are not yet ready to enable provisioning for Groups, disable that.
+    {{% /notes %}}
+
+![mappings](/images/docs/guides/scim/azuread/mappings.png)
+
+### Adjust User Attribute Mappings
+
+Update the mapping for the **userName** attribute to be sourced from **mailNickname** instead of **userPrincipalName**. In the **Mappings** expansion panel, click **Provision Azure Active Directory _Users_** and then click on the corresponding attribute mapping as shown below.
```

Comment:
@praneetloke I noticed something that I think may be an issue as I was going through this guide. Should we be suggesting `mailNickname` as the source of username? I feel like this is going to be highly prone to a lot of collisions as `mailNickname` is just the email address without the `@domain` if I understand correctly. For example `sean@pulumi.com` would result in a username of just `sean` being mapped. At smaller companies I think this firstname@domain is very common and being that this needs to be unique across _all_ Pulumi user accounts I think this may be likely to collide, but also not sure if we really have a lot of other options until we are able to implement some sort of autonaming scheme - so I am guessing this may be our best bet for now.

Reply:
@sean1588 very good point! You are correct in your understanding that `mailNickname` is the username prefix from an email address. For the situation where email addresses only use the first name of employees, what do you think about updating the doc to remind admins that an alternate exists if they prefer their users to use the format "givenName-surname"? Just like with Okta (and most IdPs I think) Azure AD also allows the user to customize the mapping of individual attributes. You could choose "Expression" to set the value to be `[givenName]-[surname]`. But for this, the first and last name attributes of users should be filled in, which in some cases _may_ not be.

![image](https://user-images.githubusercontent.com/1466314/108881096-d0f34180-75b7-11eb-91fd-3ae2bbafd12d.png)

> but also not sure if we really have a lot of other options until we are able to implement some sort of autonaming scheme

Yep I think this is what we need to do. We did do that at one point prior to having the ability to connect multiple identities. We should do this for SAML users and at the same time update the doc to let admins know that our service will do as such when it encounters a conflict.

> In Okta the username attribute can be edited manually and then re-synced if there is a conflict. In Azure, I am not so sure if this is the case.

Yeah the failed provisioning task interface in Azure AD doesn't have a way to edit the username attribute specifically for users. And the `mailNickName` being derived form the `userPrincipalName` (the email) makes that attribute not directly editable in the user profile either.